doc: Updates `delete_on_create_timeout` description as suggested and use it in the examples (#3552)
mongodb/terraform-provider-mongodbatlas: docs/guides/2.0.0-upgrade-guide.md

# MongoDB Atlas Provider 2.0.0: Upgrade and Information Guide

The Terraform MongoDB Atlas Provider version 2.0.0 has the following new features and breaking changes:

> ## New Resources, Data Sources, and Features

Reviewer: nice change making the new features and breaking changes sections have headings

> - **New `delete_on_create_timeout` attribute**: Multiple resources now support a `delete_on_create_timeout` boolean attribute that controls automatic cleanup behavior when resource creation times out. When set to `true` (default), the provider will automatically delete the partially created resource if creation times out, helping to avoid orphaned resources. When set to `false`, the resource will remain in its partially created state for manual cleanup. This attribute is available in the following resources:

Reviewer 2: partially created is misleading and also the automatic word.

```suggestion
- **New `delete_on_create_timeout` attribute**: Multiple resources now support a `delete_on_create_timeout` boolean attribute that controls cleanup behavior when resource creation times out. When set to `true` (default), the provider will delete the underlying resource if the create operation times out, helping to avoid orphaned resources. This attribute is available in the following resources:
```

Author: changed in [5791db5](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3552/commits/5791db5e04a9157be23ac0f4f16956e03cd04a63)
